### ISSUE TITLE
Source build of 1.7.0 broken Syntax errors with stats.go

### ISSUE THREAD

**Issue by prologic** [docker/docker#14112]
Just tried to build 1.7.0 and ran into this:

```#!bash
prologic@daisy
Tue Jun 23 10:37:05 
~/contrib/docker
$ sudo pkgmk -d
=======> Building '/mnt/data/cache/packages/docker#1.7.0-1.pkg.tar.gz'.
cp /mnt/data/cache/sources/check-config.sh /home/prologic/work/crux/contrib/docker/work/src
bsdtar -p -o -C /home/prologic/work/crux/contrib/docker/work/src -xf /mnt/data/cache/sources/v1.7.0.tar.gz
cp docker /home/prologic/work/crux/contrib/docker/work/src
cp docker.conf /home/prologic/work/crux/contrib/docker/work/src
+ build
+ cd docker-1.7.0
+ export AUTO_GOPATH=1
+ AUTO_GOPATH=1
+ export DOCKER_GITCOMMIT=0baf609845
+ DOCKER_GITCOMMIT=0baf609845
+ unset CC
+ unset CXX
+ ./hack/make.sh dynbinary
# WARNING! I don't seem to be running in the Docker container.
# The result of this command might be an incorrect build, and will not be
#   officially supported.
#
# Try this instead: make all
#

---> Making bundle: dynbinary (in bundles/1.7.0/dynbinary)
Created binary: bundles/1.7.0/dynbinary/dockerinit-1.7.0
Building: bundles/1.7.0/dynbinary/docker-1.7.0
# github.com/docker/docker/api/client
.gopath/src/github.com/docker/docker/api/client/stats.go:162: syntax error: unexpected range, expecting {
.gopath/src/github.com/docker/docker/api/client/stats.go:182: non-declaration statement outside function body
.gopath/src/github.com/docker/docker/api/client/stats.go:183: syntax error: unexpected }
# github.com/docker/docker/daemon
.gopath/src/github.com/docker/docker/daemon/debugtrap.go:17: syntax error: unexpected range, expecting {
=======> ERROR: Building '/mnt/data/cache/packages/docker#1.7.0-1.pkg.tar.gz' failed.
```

**Comment by prologic** [comment]
``#!bash
$ go version
go version go1.3.3 linux/amd64
```

**Comment by phemmer** [comment]
The syntax docker is using requires go 1.4: https://golang.org/doc/go1.4#forrange

Since it looks like you're trying to build a package, I would recommend reading https://github.com/docker/docker/blob/master/project/PACKAGERS.md
It covers the requirements for properly building a docker package.

**Comment by prologic** [comment]
Cool Thanks @phemmer ; btw was the requirement to bump to Go 1.4 documented anywhere? I'm not getting "packager notices" for when Docker is released. We used to get a release announcement, the sha hash and what not. Be good to also get any build changes in that announcement (*but we need to get that announcement process back*). This is the 2nd release I've had to try to hunt down important information :)
